### ISSUE TITLE
Use Glances as a NSCA client

### ISSUE THREAD

**Issue by mohierf** [nicolargo/glances#727]
As far as I understood, Glances has several running modes (standalone, server, ...) but does not propose some kind of `monitoring agent` behavior ... let me explain.

 I am currently using NSClient as a monitoring agent for my Windows hosts (more than 1000 devices). NSClient has its own checks scheduler that will run local health checks (CPU, memory, ...) on a periodical basis. Once a check run, NSClient can use NSCA protocol to send the checks results to a monitoring server like Nagios or Shinken.

 1/ I would like to replace NSClient with Glances. Each check result is sent to the monitoring server as a passive check result using NSCA (or any other protocol ...). 

 2/ I would like to use Glances as a monitoring agent that will alert my central server only when some  checks results changed. I know that I can configure Glances for running a script on warning or critical ... what about recover ? It may be interesting if this behavior would be activated on a gloabl basis. For instance, configuring a central NSCA (or HTTPS) receiver where all the checks results state change would be sent ?

 An idea to be discussed ...

**Comment by nicolargo** [comment]
Hi @mohierf , thanks for the (very interesting) feature proposal.

Let me clarify your needs:

1) You want to uses Glances as a NSCA client. Glances will send periodically (using the -t <sec> refresh command line tag) stats to a NSCA server. To do this, Glances should be modified as following:
- Create an new export mode to NSCA server. Glances will be run as a agent using the command line:

    glances -t 300 --quiet --export-nsca

with the new following configuration in the glances.conf file:

    [nsca]
    # Configuration file for the --export-influxdb option
    # https://www.nsca-ng.org/
    host=nscaserver
    port=5668
    passwork=mysecret

There are existing Python NSCA client (to be tested, it is a C binding): 
- https://github.com/weiss/nsca-ng/tree/master/python
- https://pypi.python.org/pypi/send_nsca

2) For this need, i think that you have all you need in the current Glances version.
- first off all, create a central NSCA server (https://www.nsca-ng.org/)
- on each node, install nsca_send (https://www.nsca-ng.org/documentation/send_nsca.html)
- on each node, install Glances
- define action thought the configuration file (example for memory critical). Not tested...:
   
    [mem]
    critical=90
    critical_action=echo -e "host\tstatus\tCritical memory: {{percent}}" | send_nsca -H <nscaserver> -p <port> 

- run Glances (without any GUI):

    glances --quiet

Discussion to be continued before implementation...

**Comment by mohierf** [comment]
Hi @nicolargo 
Thanks for your answer.

1/ we agree on the feature. 

 2/ what about the recovering? Is it possible to define an action when a check recovers to OK ?

**Comment by mohierf** [comment]
I made some tests with:
```
[cpu]
# Default values if not defined: 50/70/90
user_careful=50
user_warning=70
user_critical=90
#user_log=False
user_ok_action=echo 'shinkenpoller\tCpu\t0\tCPU {{user}} / {{max}}' | send_nsca -H 192.168.0.40 -c /etc/send_nsca.cfg
user_warning_action=echo 'shinkenpoller\tCpu\t1\tCPU Warning {{user}} / {{max}}' | send_nsca -H 192.168.0.40 -c /etc/send_nsca.cfg
user_critical_action=echo 'shinkenpoller\tCpu\t2\tCPU Critical {{user}} / {{max}}' | send_nsca -H 192.168.0.40 -c /etc/send_nsca.cfg
```

What I got in Shinken:
```
[1448641923] PASSIVE SERVICE CHECK: shinkenpoller;Cpu;1;CPU Warning 85.7 /
[1448641923] PASSIVE SERVICE CHECK: shinkenpoller;Cpu;0;CPU 50.0 /
[1448641923] EXTERNAL COMMAND: [1448641922] PROCESS_SERVICE_CHECK_RESULT;shinkenpoller;Cpu;1;CPU Warning 85.7 /
[1448641923] EXTERNAL COMMAND: [1448641922] PROCESS_SERVICE_CHECK_RESULT;shinkenpoller;Cpu;0;CPU 50.0 /
```

**One question and some remarks:**
- where can I find an exhaustive list of the variables that may be used as {{mustache}}?

- when Glances is starting the notifications should be delayed because there is always a Cpu overload on start and, as of it, Warning/Critical alerts are raised

**Comment by nicolargo** [comment]
"where can I find an exhaustive list of the variables that may be used as {{mustache}}?"

=> Here: https://github.com/nicolargo/glances/wiki/The-Glances-2.x-API-How-to

"when Glances is starting the notifications should be delayed because there is always a Cpu overload on start and, as of it, Warning/Critical alerts are raised"

=> Just added a new issue for the next release: https://github.com/nicolargo/glances/issues/732

**Comment by mohierf** [comment]
Ok for the link to API ... I already got it. But it is not that easy to find out what can be used in the commands ... for example: I tried {{hostname}} but with no success despite it is avaiable in the stats ...

**Comment by nicolargo** [comment]
For the moment, you can only access to attribute inside the plugin area. For exemple inside the [mem] section, you can access to: '{"**available**": 5266268160, "**used**": 3007328256, "**cached**": 2949533696, "**percent**": 36.3, "**free**": 5266268160, "**inactive**": 2034712576, "**active**": 3497422848, "**total**": 8273596416, "**buffers**": 183599104}'

I just add an enhancement request to allow user to use global attributes like: hostname... (list to be defined) => https://github.com/nicolargo/glances/issues/733
